group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1244,15 +1244,16 @@ private void throwIfClassicProtocolUnmatched(
         String requestProtocolType,
         String requestProtocolName
     ) {
+        String protocolName = member.supportedClassicProtocols().get().iterator().next().name();
```

Comment:
I just recalled that the protocol type and name are optional in the sync group request so we should only validate them if they are non-null.

Reply:
I'm not sure if it's necessary as we've checked if the member uses the classic protocol before, but I guess it doesn't hurt to explicitly check the protocols here. Let me add the validation..

Reply:
What I said is different. What I meant is that requestProtocolType and requestProtocolName could be null here.

Reply:
Ah right, they should be validated

Reply:
Yes, if `requestProtocolType` or `requestProtocolName` is null, the validation will fail.

> we should only validate them if they are non-null.
We do validate them even if they are null in the existing classic group sync to the classic group https://github.com/apache/kafka/blob/056d232f4e28bf8e67e00f8ed2c103fdb0f3b78e/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java#L4335-L4342 so I think we should also fail the validation if either of them is null.

Reply:
Ah now I get it.. Let me add the check

Reply:
You may be able to reuse `isProtocolInconsistent`.